### PR TITLE
Fixup prompt for ip

### PR DESCRIPTION
**Pull request by dennisjbell**
Adds ip validation to prompt-for line.

Also fixes issues with hooks/new script that use `set -eu` and adds the missing --echo option for the secret-line prompt.

### CODE REVIEW

File: t/param-asking.t [hunk 1]
```diff
@@ -349,6 +349,53 @@ expect_ok "URL parsing - finish up", $cmd, [
...
+];
+expect_ok "IP parsing - too many octets", $cmd, [
+  "Specify the path", sub {
+    fail "Invalid IP address not correctly identified: too many octets";
+  }], [
+  "123.4.5.6.78 is not a valid IPv4 address[\r\n]{1,2}>", sub {
+    $_[0]->send("123.4.5.67\n");
+  }
+];
+
```
Comment (jhunt) [inline t/param-asking.t]: Can we add a "valid" test for something like 10.0.0.1 and 255.0.0.4?

Comment (dennisjbell) [reply t/param-asking.t]: Yup, will do

Comment (dennisjbell) [reply t/param-asking.t]: Ready for your re-review...

Comment (dennisjbell) [reply t/param-asking.t]: Added bounds checks

**Pull request merged**